### ISSUE TITLE
[Component] Textlink

### ISSUE THREAD

**Issue by thomas-leguellec** [adevinta/spark-ios#522]

**Comment by elisa-hery** [comment]
<google-sheets-html-origin>

Testing informations |  
-- | --
Risk level | high (= must be tested by QAA before release)
Does this ticket need to be design reviewed ? | no ?
Referent designer : |  ?
Spec : | [figma](https://www.figma.com/file/0QchRdipAVuvVoDfTjLrgQ/Spark-Specs-and-Handoff?type=design&node-id=8638-1179&mode=design&t=mKntrlYcJUfZ3YIo-0)
UATs : | [confluence](https://confluence.ets.mpi-internal.com/display/SPAR/Textlink)
configurator : | [confluence](https://confluence.ets.mpi-internal.com/display/SPAR/Progress+tracker#Textlink-demo-app-configurator)

**Comment by robergro** [comment]
To test the component, you can use this branch: 
https://github.com/adevinta/spark-ios/tree/component/textlink/all

**Comment by elisa-hery** [comment]
@robergro here are my questions /comments : 
- default color for textLink is `basic`, I think it should be the same on demo app, WDYT ?
- I have a bug when opening textlink on iPad : 

https://github.com/adevinta/spark-ios/assets/142396647/e1670dde-d81a-4339-8bfd-80ee16edbb18


- out of curiosity, why was the line limit necessary (since it's not in the specs) ? It's not really compatible with dynamic types since it hides almost the hole text : 
<img width="301" alt="Capture d’écran 2024-01-15 à 16 05 27" src="https://github.com/adevinta/spark-ios/assets/142396647/0b617d8c-b82f-47d6-b4c1-cde9740866ef">

- why is there always a block on the textLink's right on demo app ? 
- out of curiosity, what is `control Type` ?
- note : I cannot really test the functional part (link to an anchor or redirection)

**Comment by robergro** [comment]
**- default color for textLink is basic, I think it should be the same on demo app, WDYT ?**
You're right, I change to .basic

**- I have a bug when opening textlink on iPad :**
Should be fixed on this [PR](https://github.com/adevinta/spark-ios/pull/749) 

**- out of curiosity, why was the line limit necessary (since it's not in the specs) ? It's not really compatible with dynamic types since it hides almost the hole text :**
It is a specificity of the label on iOS, there is a possibility to set a number of line limitation (0 mean illimited). I just added that for our consumer who would like to be sure to have X lines

**- why is there always a block on the textLink's right on demo app ?**
The component takes the all available space left. So I we do not display the right space, the text link with a little text will takes the 100% of the screen.
If we add a right space, the text link will takes only the required width.
BUT if we have a long text like a paragraph, the right space will have a 0px width, because on the demo app, the priority is for the component (and not for the spacing) I added a screenshot with a orange background on the text link to show you the width of the textlink


**- out of curiosity, what is control Type ?**
There is three possibilities to informe the consumer that the text link is tapped:
- With a publish value
- With a action 
- With a target
The consumer must choose one of them to be informed when a tap is made.
For the demo app, we added the Control Type enum to test if the tap goes back well for all possibilities (And we show an alert)


**- note : I cannot really test the functional part (link to an anchor or redirection)**
It is not our role to manage the redirection. We just send the information to the parent when the user tap on the text link. (As we do for all controllable component)

Without right space (the width of the textlink takes 100% of the parent width):
![Image](https://github.com/adevinta/spark-ios/assets/6044208/88ff1aa2-39aa-42c6-8da2-532fd5a377b9)

With right space (the size of the textlink depends on its content):
![Image](https://github.com/adevinta/spark-ios/assets/6044208/f15dca13-2a7e-496c-965d-66bcf19cce55)
